ViewControllers/DetailViewController.swift

```diff
@@ -202,40 +238,67 @@ class DetailViewController: UIViewController {
-            titleLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            titleLabel.topAnchor.constraint(equalTo: contentView.layoutMarginsGuide.topAnchor),
+            titleLabel.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
+            titleLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
 
             imageView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 16),
             imageView.centerXAnchor.constraint(equalTo: titleLabel.centerXAnchor),
-            imageView.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 0.61),
+            imageView.widthAnchor.constraint(equalTo: contentView.widthAnchor, multiplier: 0.61),
             imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 1.41),
             
-            releaseStack.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
+            releaseStack.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
             releaseStack.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 35),
 
             durationStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             durationStack.topAnchor.constraint(equalTo: releaseStack.bottomAnchor, constant: 8),
 
             movieDescriptionLabel.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            movieDescriptionLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            movieDescriptionLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
             movieDescriptionLabel.topAnchor.constraint(equalTo: durationStack.bottomAnchor, constant: 16),
 
             collectionView.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             collectionView.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
             collectionView.rightAnchor.constraint(equalTo: movieDescriptionLabel.rightAnchor),
-            collectionView.heightAnchor.constraint(equalToConstant: 200),
+            collectionView.heightAnchor.constraint(equalToConstant: calculateCollectionViewHeight())
         ])
     }
     
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        scrollView.contentSize = CGSize(width: view.frame.size.width, height: calculateScrollViewContentSize())
+    }
+    
+    private func calculateCollectionViewHeight()-> CGFloat {
+        var languageRows = languages.count / 4
+        languages.count % 4 != 0 ? languageRows += 1 : nil
+        let languageRowsHeight = (languageRows * 27) + ((languageRows - 1) * 4)
+        
+        var genreRows = genres.count / 4
+        genres.count % 4 != 0 ? genreRows += 1 : nil
+        let genreRowsHeight = (genreRows * 27) + ((genreRows - 1) * 4)
+        
+        return CGFloat(languageRowsHeight + 16 + genreRowsHeight)
```

Comment:
If we choose to calculate the collection height manually we should be using the content that is displayed in the collection instead of languages and rows lists. Yes, they are being displayed here, but it can happen that it's not in sync with what is actually being displayed which will lead to wrong sizes. We can use the collection datasource methods here to get number of sections and items:

```swift
var totalHeight: CGFloat = 0
let numberOfSections = collectionView.numberOfSections
for section in 0..<numberOfSections {
    let numberOfItems = collectionView.numberOfItems(inSection: section)
    // calculate section height here and add to "totalHeight"
}
return totalHeight
```
Additionally, this code is generic so you can reuse it in any collection view regardless of the content